public/tabs/chat/chat_page_content.tsx

```diff
@@ -44,12 +44,27 @@ export const ChatPageContent: React.FC<ChatPageContentProps> = React.memo((props
   const chatActions = useChatActions();
   const registry = getIncontextInsightRegistry();
   const configSchema = getConfigSchema();
+  const latestInputRef = useRef<HTMLDivElement>(null);
+  const [messageSpacerHeight, setMessageSpacerHeight] = useState(0);
 
   useLayoutEffect(() => {
-    if (!props.chatScrollTopRef.current) {
+    if (latestInputRef.current) {
+      setMessageSpacerHeight(window.innerHeight - latestInputRef.current.clientHeight - 135);
+
+      setTimeout(() => {
+        // Do the scrolling in the next loop to avoid page flashing
+        if (latestInputRef.current) {
+          latestInputRef.current.scrollIntoView({ behavior: 'smooth' });
+        }
+      }, 0);
+    }
+  }, [loading, latestInputRef, window.innerHeight]);
```

Comment:
I am thinking if we should add `latestInputRef.current.clientHeight`.

Reply:
I am not sure that `latestInputRef.current`is possibly null so should we put it into dependency list?

Reply:
Forget it, it should be fine.